File: src/main/java/io/dustin/finstage/common/http/RestTemplateHttpClient.java
package io.dustin.finstage.common.http;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;
import org.springframework.web.client.RestTemplate;

@Component
@RequiredArgsConstructor
public class RestTemplateHttpClient implements HttpClient {

    private final RestTemplate restTemplate;

    @Override
    public <T> T get(String url, Class<T> responseType) {
        return restTemplate.getForObject(url, responseType);
    }
}

File: src/main/kotlin/io/dustin/finstage/common/http/RestTemplateHttpClient.kt
package io.dustin.finstage.common.http

import org.springframework.stereotype.Component
import org.springframework.web.client.RestTemplate

@Component
class RestTemplateHttpClient(
        private val restTemplate: RestTemplate
) : HttpClient {

    override fun <T> get(url: String, responseType: Class<T>): T? {
        return restTemplate.getForObject(url, responseType)
    }
}
